sympy/physics/mechanics/functions.py

```diff
@@ -422,8 +444,17 @@ def find_dynamicsymbols(expression, exclude=None):
             raise TypeError("exclude kwarg must be iterable")
     else:
         exclude_set = set()
-    return set([i for i in expression.atoms(AppliedUndef, Derivative) if
-            i.free_symbols == t_set]) - exclude_set
+    try:
+        if isinstance(expression, Vector):
+            expr=expression.to_matrix(reference_frame)
+            return set([i for i in expr.atoms(AppliedUndef, Derivative) if
+                    i.free_symbols == t_set]) - exclude_set
+    except TypeError:
```

Comment:
I'm not sure this is the best check here. Why not check explicitly? Like "if isinstance(expression, Vector) and reference_frame=None".